zombsole/gym_env.py

```diff
@@ -1,15 +1,108 @@
 #!/usr/bin/env python
 # coding: utf-8
 from os import path, system
+from abc import ABC, abstractmethod
+from typing import Tuple
 from gym.core import Env
 from gym.spaces import Box
 from gym.spaces.discrete import Discrete
 from zombsole.game import Game, Map
 from zombsole.renderer import NoRender
+from zombsole.players.agent import Agent
 import time
 import numpy as np
 
 
+class SinglePlayerObservation(ABC):
+    @abstractmethod
+    def get_observation(self, game: Game):
+        pass
+    
+    @abstractmethod
+    def get_observation_space(self):
+        pass
+
+
+class WorldSimpleObservation(SinglePlayerObservation):
+    def __init__(self, map_size: Tuple[int, int]):
+        self.map_size = map_size
+
+    def get_observation(self, game: Game):
+        observation = np.array(game.encode_world_simple())
+        return observation.reshape( (1,) + observation.shape )
+
+    def get_observation_space(self):
+        return Box(low=0, high=8*16*16, shape=(1, self.map_size[1], self.map_size[0]), dtype=np.int32)
+
+
+class WorldChannelsObservation(SinglePlayerObservation):
+    def __init__(self, map_size: Tuple[int, int]):
+        self.map_size = map_size
+
+    def get_observation(self, game: Game):
+        return np.array(game.encode_world_with_channels())
+
+    def get_observation_space(self):
+        return Box(low=0, high=128, shape=(self.map_size[1], self.map_size[0], 3), dtype=np.int32)

```

Comment:
For consistency, we might need to permute the order of the dimensions so that the channels are associated with the first index.